packages/gensx-storage/src/blob/provider.tsx

```diff
@@ -0,0 +1,39 @@
+import { Component } from "@gensx/core";
+
+import { BlobContext } from "./context.js";
+import { FileSystemBlobStorage } from "./filesystem.js";
+import { RemoteBlobStorage } from "./remote.js";
+import { BlobProviderProps } from "./types.js";
+
+/**
+ * BlobProvider component that provides blob storage to its children
+ *
+ * @example
+ * ```jsx
+ * // Use local filesystem storage
+ * <BlobProvider kind="filesystem" rootDir="/tmp/blob-storage">
+ *   <YourComponent />
+ * </BlobProvider>
+ *
+ * // Use cloud storage
+ * <BlobProvider kind="cloud">
+ *   <YourComponent />
+ * </BlobProvider>
+ * ```
+ */
+export const BlobProvider = Component<BlobProviderProps, never>(
+  "BlobProvider",
+  (props) => {
+    // Create the appropriate storage implementation based on kind
+    if (props.kind === "filesystem") {
```

Comment:
We haven't talked about it, but would it make sense to have a `"memory"` version? I'm on the fence.

Reply:
Could you say more about what you're thinking for `"memory"`?

Reply:
In-memory could be useful for the local dev server, but if you're starting/stopping your server or if you're just running the index.ts file it is much less useful.